htdocs/compta/paiement.php

```diff
@@ -236,6 +236,14 @@
 		    }
 	    }
 
+        foreach ($amounts as $key => $value)	// How payment is dispatch
+        {
+            $invoice = new Facture($db);
+            $invoice->fetch($key);
+            if ($invoice->paye) unset($amounts[$key]);
```

Comment:
May be a warning report would be better ?
    		setEventMessages(..., ...,  'warnings');

If we just clean, it mean, user may ask a payment of 100 euros on 4 invoices of 25, and at the end dolibarr record only 75 euros because the invoice 4 was already paid. Users has made a check of 

Also sometimes, we pay twice. it was an error, but we should be able to record it in dolibarr that we paid more/twice.

Reply:
I replaced it by a seteventmsg, but to me, if an invoice is paid, we shouldnt be able to add a paiement.

Reply:
I agree that we should not be able to enter a payment if invoice is paid.
But your first submission has effect to introduce a bug.
My suggestion was to refuse the payment with a warning to explain why payment is refused, not forcing data to another value to force the validation of payment.